deepchem/models/losses.py

```diff
@@ -465,6 +468,130 @@ def loss(inputs):
         return loss
 
 
+class GroverPretrainLoss(Loss):
+
+    def _create_pytorch_loss(self):
+        import torch.nn as nn
+
+        def loss(preds,
+                 targets,
+                 weights: Optional[List[Sequence]] = None,
+                 dist_coff=0.1):
+            """
+            Parameters
+            -----------
+            preds: torch.Tensor
+                predictions
+            targets: torch.Tensor
+                targets (real values)
+            dist_coff: float, default 0.1
+                loss metric
+
+            Returns
+            -------
+            loss: torch.Tensor
+                loss value
+            """
+            av_task_loss = nn.NLLLoss(ignore_index=0,
+                                      reduction="mean")  # same for av and bv
+
+            fg_task_loss = nn.BCEWithLogitsLoss(reduction="mean")
+            # av_task_dist_loss = nn.KLDivLoss(reduction="mean")
+            av_task_dist_loss = nn.MSELoss(reduction="mean")
+            fg_task_dist_loss = nn.MSELoss(reduction="mean")
+            sigmoid = nn.Sigmoid()
+            av_atom_loss, av_bond_loss, av_dist_loss = 0.0, 0.0, 0.0
+            fg_atom_from_atom_loss, fg_atom_from_bond_loss, fg_atom_dist_loss = 0.0, 0.0, 0.0
+            bv_atom_loss, bv_bond_loss, bv_dist_loss = 0.0, 0.0, 0.0
+            fg_bond_from_atom_loss, fg_bond_from_bond_loss, fg_bond_dist_loss = 0.0, 0.0, 0.0
+
+            if preds["av_task"][0] is not None:
```

Comment:
Instead of assuming specific structure on `preds`, I'd suggest just accepting different tensors for the different sub-loss terms.

Reply:
For example, you can have atom, bond, functional group terms